src/language/classScope.ts

```diff
@@ -0,0 +1,108 @@
+/**
```

Comment:
I don't think this file belongs here...
I know @adidahiya suggested moving it, but this is not reusable at all.
I'd prefer to include it in the rule's source file. It will only get you to a total of about 300 lines, which is still reasonable.